odl/set/sets.py

```diff
@@ -282,6 +298,11 @@ def contains_set(self, other):
                 isinstance(other, RealNumbers) or
                 isinstance(other, Integers))
 
+    def contains_all(self, array):
+        """Test if `array` is an array of real or complex numbers."""
+        array = np.asarray(array)  # allow seqence type input
+        return is_scalar_dtype(array.dtype)
```

Comment:
Dislike having to do a cast. Why cant we (somehow) learn what dtype it is without creating new memory?